### ISSUE TITLE
What LR should be chosen when replacing AdamW with ADOPT

### ISSUE THREAD

**Issue by omrastogi** [iShohei220/adopt#9]
I am training PixArt. 
Training with AdamW, a regime of lr=4e-4 and weight_decay=1e-4 with linear decay works for the model. However, when exchanged with ADOPT the gradient exploded at just 40th iteration. 
![image](https://github.com/user-attachments/assets/7d8b3eb5-0b0b-42a7-912e-ee51dec8ab11)

Since the paper claims that no hyperparameter tuning is required, I am assuming that there must be an optimal new LR for ADOPT. Can you please help me with this?

**Comment by iShohei220** [comment]
Thank you for your report.

Based on the feedbacks from some practitioners, we slightly modify the implementation, and fixed the instability in the commit on Nov 22 (see README for more details), so please try the updated version.
If it is still unstable, please report it again.

Thanks!
